x/gamm/pool-models/balancer/pool_suite_test.go

```diff
@@ -1095,4 +1095,36 @@ func (suite *KeeperTestSuite) TestRandomizedJoinPoolExitPoolInvariants() {
 	for i := 0; i < 50000; i++ {
 		testPoolInvariants()
 	}
+	testCaseCorners := []testCase{
+		{ //join pool for values greater than in the pool
+			initialTokensDenomIn:  1,
+			initialTokensDenomOut: 9_223_372_036_854_775_807,
+			percentRatio:          101,
+		},
+		{ //with percentRatio = 0
+			initialTokensDenomIn:  1,
+			initialTokensDenomOut: 9_223_372_036_854_775_807,
+			percentRatio:          0,
+		},
+		{ //
+			initialTokensDenomIn:  9_223_372_036_854_775_807,
+			initialTokensDenomOut: 1,
+			percentRatio:          101,
+		},
+		{
+			initialTokensDenomIn:  9_223_372_036_854_775_807,
```

Comment:
asking to learn: is there a specific reason for this number? or does it just need to be big enough? if latter, big enough compared to what?

Reply:
it is the maximum number of type int64, I think the edge cases are the ones that lie at border